Filipo0on/chessgame
# Feature/apollo express

**related issue:** feature/apollo-server-graphQL

**Important notice**
Apollo Server is no longer connected or used in this branch, most due to lacking documentation och the Apollo Server. Instead, I'm using express and express-graphql packages are being used to handle the server.

DO NOT merge this before the update-structure PR.

Also this PR does not include any manipulations, only queries.

**Features completed**
There is a server that can be started and basic schemas for graphQL.

**Dependencies installed (that are in active use)**
[express](https://www.npmjs.com/package/express)
[express-graphql](https://www.npmjs.com/package/express-graphql)
[json-server](https://www.npmjs.com/package/json-server)
[axios](https://www.npmjs.com/package/axios)
[graphql](graphql)

**To test this branch**
1. run `npm install -g json-server`
2. in server folder run `npm install`
_you do not need to do npm start or bother with client side for this PR_
3. To start the server run `node index.js` in the terminal, make sure that you are in the server folder!
4. To start json-server run `json-server -p 1337 --watch db.json` run this in a different terminal so you have to separate terminals for server and json-server.
5. To see the rest API go to http://localhost:1337/games or http://localhost:1337/players, img below displays the player rest
![playerrest](https://user-images.githubusercontent.com/11257031/50683251-af1e5580-1011-11e9-9ef5-c48963c5e298.PNG)

6. To see the graphiql playground go to http://localhost:4000/graphql
![graphiql](https://user-images.githubusercontent.com/11257031/50683273-c6f5d980-1011-11e9-8822-62dae74cccbf.PNG)

**How to run queries**
There are 4 queries that you can run at this point in time.
1. Query all games
`getGames{
    id
    gameType
    gameTime
    gameAddTime
    gameStarted
  }`
2. Query game based on ID
`getGame(id: 1){
    id
    gameType
    gameTime
    gameAddTime
    gameStarted
  }`
3. Query all players
`getPlayers {
    id
    userName
  }`
4. Query player based on ID
`getPlayer(id: 1) {
    id
    userName
  }`

We will make a little lesson about running queries on Monday as well!
